### PR TITLE
Added POST /api/admin/event/ to create a new event

### LINKED ISSUE #380: Backend - Endpoint that adds an event
**Is your feature request related to a problem? Please describe.**
There is no easy way for admins to add events. For this to be implemented in frontend, an endpoint is needed in the backend.

**Describe the solution you'd like**
Add an endpoint to the admin controller that lets the admin create an event with this provided input:
- DeliveryAddress (general address that the givers should deliver to)
- StartDate (date the event will be active from - must be after the current date)
- EndDate (date the event will be active to)

**Describe alternatives you've considered**
There is another solution that doesn't require the startDate as input. The startDate will then be the current date.

### PR DESCRIPTION
**Pull request by HGodal**
Fixes #380 
Mentions #374 

Creating a POST route for /api/admin/event/ to create a new event, or update one that already exists.

### CODE REVIEW

**Review by Sommerli — CHANGES_REQUESTED** [review]
Nice work 👍 
Left a small suggestion for the controller method

Also i request clarification on the Event-class, as its keys has been swapped.

File: GiEnJul/Entities/Event.cs [hunk 1]
```diff
@@ -4,12 +4,17 @@ namespace GiEnJul.Entities
 {
     public class Event : EntityBase
     {
-        //RK - Location
-        //PK - Event name
+        //RK - EventType
+        //PK - City
```
[the remark this answers is not in the document]
Comment (HGodal) [reply GiEnJul/Entities/Event.cs:8]: Yes 😅 I'll revert the change

File: GiEnJul/Controllers/AdminController.cs [hunk 2]
```diff
@@ -103,6 +103,30 @@ public async Task<IList<GetConnectionDto>> GetConnectionsByLocationAsync(string
             }
             return connecitonDtos;
         }
+
+        [HttpPost("event")]
+        [Authorize(Policy = "ReadConnection")]
+        public async Task<ActionResult> PostEventAsync([FromBody] PostEventDto eventDto)
+        {
+
+            if (eventDto.StartDate <= System.DateTime.Today || eventDto.StartDate >= eventDto.EndDate)
```
[the remark this answers is not in the document]
Comment (HGodal) [reply GiEnJul/Controllers/AdminController.cs:112]: startDate <= today throws an error if the start date is set to an earlier day than today. The criterion allows events to start at a later time.

[the remark this answers is not in the document]
Comment (HGodal) [reply GiEnJul/Controllers/AdminController.cs:112]: > Dont know if there will be use cases where events are created with an upcoming startdate (startDate > DateTime.Today). We should ask what the admin thinks :)

startDate <= today throws an error if the start date is set to an earlier day than today. The criterion allows events to start at a later time.

File: GiEnJul/Entities/Event.cs [hunk 3]
```diff
@@ -4,12 +4,17 @@ namespace GiEnJul.Entities
...
 
+        public string EventType { get; set; }
```
[the remark this answers is not in the document]
Comment (HGodal) [reply GiEnJul/Entities/Event.cs:10]: I'll revert the name back to EventName to avoid confusion.

**Review by TTorgersen — APPROVED** [review]
Nice fixes:)

**Review by Sommerli — APPROVED** [review]

**Pull request merged**